openedx/openedx-platform
# [BD-13] refactor: deprecates ModuleSystem properties related to code sandboxing

## Description

Builds on the shim added by https://github.com/edx/edx-platform/pull/29190 to deprecate the following `ModuleSystem` attributes which are only used by CAPA problems:
* `cache`: deprecated in favour of the added CacheService
* `can_execute_unsafe_code`, `get_python_lib_zip`: deprecated in favour of the added SandboxService

Also fixes the [`safe_exec.tests.TestSafeOrNot.test_cant_do_something_forbidden`](https://github.com/edx/edx-platform/blob/78799f25ee6f0015c5288174602c29eea6de706a/common/lib/capa/capa/safe_exec/tests/test_safe_exec.py#L84) test , which was failing on my devstack with latest master. But since Jenkins and the github CI don't run in a virtualenv, this test is skipped, so this issue had gone unnoticed.

This change is only a refactoring, and should not affect Learners, Course Authors, or anyone else using edx-platform.

## Supporting information

* [Content Core Platform Simplication project](https://openedx.atlassian.net/wiki/spaces/AC/pages/3091629503/Content+Core+Platform+Simplification)
* [BLENDED-113](https://openedx.atlassian.net/browse/BLENDED-113) (edX internal ticket)
* [FAL-2209](https://tasks.opencraft.com/browse/FAL-2209) (OpenCraft internal ticket)

## Testing instructions

* LMS: https://pr29260.sandbox.opencraft.hosting/
* Studio: https://studio.pr29260.sandbox.opencraft.hosting/

LMS:
1. Login as a demo learner (`honor@example.com` / `edx`)
1. Enrol in the [BD-13 test course](https://pr29260.sandbox.opencraft.hosting/courses/course-v1:OpenCraft+BD-13+2021_1/about)
1. Navigate to the [Complex Blocks > Custom python problem](https://discovery.pr29260.sandbox.opencraft.hosting/learning/course/course-v1:OpenCraft+BD-13+2021_1/block-v1:OpenCraft+BD-13+2021_1+type@sequential+block@862667c509144a4c83381589185f9072/block-v1:OpenCraft+BD-13+2021_1+type@vertical+block@0edf5722fa1240ef82228ece3e68adcc). Ensure that you can answer the question correctly (this tests the python code sandboxing).
1. Switch to the Legacy Course Experience and ensure you can still answer this problem.

Regression testing: 

1. Browse around the Demo course, paying particular attention to the CAPA problems.
1. Check that you can submit answers to problems, and save changes to blocks. (This tests that these changes haven't broken anything else with CAPA problems.)
1. Switch to the Legacy Course Experience and ensure you can submit answers to problems there too.

Studio:
1. Login as a demo staff (`staff@example.com` / `edx`)
2. Navigate to the Demo course and browse through the blocks, paying particular attention to the CAPA problem blocks. Ensure they render and edit as expected.
3. Check that you can submit answers to problems, and save changes to blocks.
4. Preview unpublished changes and ensure they render as expected.
4. Publish changes, and ensure they render in the LMS as expected.

safe_exec:

In your devstack's `lms-shell`, run the following command, and ensure all tests pass.

```bash
pytest common/lib/capa/capa/safe_exec/tests/test_safe_exec.py
```

## Deadline

None
